src/inc/sift-events/sift-events.php

```diff
@@ -674,22 +739,39 @@ public static function update_or_create_order( string $order_id, \WC_Order $orde
 	 * @return void
 	 */
 	public static function transaction( \WC_Order $order, string $status, string $transaction_type ) {
-
-		if ( ! Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$transaction ) ) {
+		// $transaction requires a positive amount, so we don't send it for free orders.
+		if ( self::is_free_order( $order ) || ! Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$transaction ) ) {
 			return;
 		}
 
+		// Determine user and session context.
+		$user_id    = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
+		$user_id    = self::format_user_id( intval( $user_id ) );
+		$user_email = wp_get_current_user()->user_email ?? $order->get_billing_email() ?? null;
+
+		// If there is no user ID, fall back to the billing email as the user ID.
+		if ( ! $user_id ) {
+			$user_id = $user_email;
+		}
+
 		$properties = array(
-			'$user_id'            => self::format_user_id( $order->get_user_id() ),
+			'$user_id'            => $user_id,
+			'$user_email'         => $user_email,
 			'$session_id'         => \WC()->session?->get_customer_unique_id() ?? '',
-			'$amount'             => self::get_transaction_micros( floatval( $order->get_total() ) ), // Gotta multiply it up to give an integer.
+			'$amount'             => self::get_transaction_micros( floatval( $order->get_total() ) ),
 			'$currency_code'      => $order->get_currency(),
 			'$order_id'           => (string) $order->get_id(),
 			'$transaction_type'   => $transaction_type,
 			'$transaction_status' => $status,
 			'$time'               => intval( 1000 * microtime( true ) ),
 		);
 
+		// If the session ID is empty, create one from the order ID.
+		// This ensures we always have a session ID even for admin-created orders.
+		if ( empty( $properties['$session_id'] ) ) {
+			$properties['$session_id'] = md5( 'order_session_' . $order->get_id() );
```

Comment:
That is linked to my previous comment: if the session is not available, `wp_get_current_user`will fail.
Regarding this specific change, though, why do we need the session ID? It seems unecessary and will not make the signal more meaningul I believe. Happy to discuss about this.

Reply:
See above.